### PR TITLE
 Support start, add testcase, and can be executed properly

### PR DESCRIPTION
**Pull request by haxxpop**
The following is the list of what I have done to add start functionality

- I have changed the Ast module and Kernel module to include the start function index.
- Have the start function run as soon as the module is loaded.
- In the print_module function in host/print.ml, I have added the start function to be the output to the screen
- Add test/start.wast to test the start function

Please consider if we should add this feature to the main repo

### CODE REVIEW

File: ml-proto/host/print.ml [hunk 1]
```diff
@@ -27,6 +27,10 @@ let print_export_sig m prefix n f =
 let print_table_elem i x =
   printf "table [%d] = func %d\n" i x.it
 
+let print_start start =
+  match start with
+  | Some x -> printf "start = func %d\n" x.it
+  | None -> printf "there is no start\n"
```
Comment (rossberg-chromium) [inline ml-proto/host/print.ml]: Nit: just don't print anything in this case.

File: ml-proto/spec/eval.ml [hunk 2]
```diff
@@ -313,12 +313,18 @@ let add_export funcs ex =
...
-   host}
+  let {memory; funcs; exports; start; _} = m.it in
+  let instance =
+    {module_ = m;
+     imports;
+     exports = List.fold_right (add_export funcs) exports ExportMap.empty;
+     memory = Lib.Option.map init_memory memory;
+     host}
+  in
+  ignore (match start with
```
Comment (rossberg-chromium) [inline ml-proto/spec/eval.ml]: Nit: Apply `ignore` more locally to the `eval_func` call.

Comment (haxxpop) [inline ml-proto/spec/eval.ml]: What does "more locally" mean ?
Does it mean something like this ?

  (match start with 
    | Some x -> ignore (eval_func instance (lookup "function" funcs x) []) 
    | None -> ());

Comment (rossberg-chromium) [inline ml-proto/spec/eval.ml]: Yes. In fact, you can also simplify the whole thing to
```
Lib.Option.app (fun x -> ignore (eval_func instance (lookup "function" funcs x) []) start;
```

**Comment by haxxpop** [comment]
@rossberg-chromium 
I have simplified both print_start and eval_func. You can check it in the latest commit.

**Comment by rossberg-chromium** [comment]
Thanks, LGTM.

**Comment by jfbastien** [comment]
Will this assert out if `$foo` isn't nullary in `(start $foo)`?

**Comment by haxxpop** [comment]
@jfbastien Oh, thank you. I have recently modified it to assert that it's nullary before calling eval_func.

**Comment by jfbastien** [comment]
Are there tests in the spec to validate this? Or do we just not do negative tests at the moment?

**Comment by haxxpop** [comment]
No, not yet. I will do it soon :)

**Comment by haxxpop** [comment]
@jfbastien I added it already. There are three negative tests in test/start.wast now.

**Comment by haxxpop** [comment]
@rossberg-chromium fixed :)

**Comment by rossberg-chromium** [comment]
LGTM, thanks!

**Comment by lukewagner** [comment]
Nice!

**Pull request merged**